api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/ContrastAccountServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+        AccountValidationResponse validationResponse = validateRequestData(accountData);
+        if (validationResponse.getValidationStatus().equalsIgnoreCase(FAILURE)) {
+            LOGGER.info("Validation failed due to missing parameters");
+            return validationResponse;
+        }
+        return getOrganizationName(accountData);
+    }
+
+    private AccountValidationResponse getOrganizationName(CreateAccountRequest pluginData) {
+        AccountValidationResponse validationResponse = new AccountValidationResponse();
+        validationResponse.setValidationStatus(FAILURE);
+        validationResponse.setMessage(VALIDATION_FAILED);
+        try {
+            String apiUrl = new URIBuilder(String.format(CONTRAST_URL_TEMPLATE, pluginData.getContrastEnvironmentName()))
+                    .setPath(ORGANIZATIONS_URL_PATH + pluginData.getContrastOrganizationId()).build().toString();
+            String authKey = Base64.getEncoder().encodeToString((pluginData.getContrastUserId() + ":"
+                    + pluginData.getContrastServiceKey()).getBytes());
+            HttpGet request = new HttpGet(apiUrl);
+            request.setHeader(HttpHeaders.AUTHORIZATION, authKey);
+            request.setHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+            request.setHeader("API-Key", pluginData.getContrastApiKey());
+            try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                try (CloseableHttpResponse response = httpClient.execute(request)) {
+                    int statusCode = response.getStatusLine().getStatusCode();
+                    if (statusCode != 200) {
+                        String errorMessage = handleErrorResponse(response.getStatusLine().getStatusCode());
+                        validationResponse.setErrorDetails(errorMessage);
+                        return validationResponse;
+                    }
+                    JsonNode jsonResponse = objectMapper.readTree(response.getEntity().getContent());
+                    String organizationName = jsonResponse.get("name").textValue();
+                    if (StringUtils.isEmpty(organizationName)) {
+                        validationResponse.setErrorDetails("Couldn't read Organization name");
+                        return validationResponse;
+                    }
+                    pluginData.setContrastOrganizationName(organizationName);
+                    validationResponse.setValidationStatus(SUCCESS);
+                    validationResponse.setErrorDetails(null);
+                    validationResponse.setMessage("Connection established successfully");
+                    return validationResponse;
+                }
+            }
+        } catch (UnknownHostException e) {
+            LOGGER.error("Unknown host error: " + e.getMessage(), e);
+            validationResponse.setErrorDetails("The Environment name may be incorrect or the server may be down.");
```

Comment:
I was trying to add more specific details to ErrorDetails field, errorMessage field now has value as "Contrast validation Failed"

Reply:
we cannot say word "may be" - we have to be 100% sure. Since env name can be wrong - find a way to check it without causing exception. I bet its possible.

Reply:
So before trying to do http request make sure that API endpoint exists. Of if http client rises specific type of errors in such case - add that specific type here so we can say it with 100% confidence that env name is wrong. And of course we cannot say that server may be down - this is complete guess. If URL is unreachable - it's just wrong for us. Let me know if it's confusing - we can chat about it during the meeting.